### PR TITLE
Allow keying properties when selecting multiple nodes

### PR DESCRIPTION
**Pull request by mihe**
https://github.com/godotengine/godot/assets/4884246/fe9cb41d-ab80-4aa4-a979-d87924c41132

Currently it's not possible to key properties for animation tracks when having multiple nodes selected in the scene tree, as pointed out among the many quality-of-life improvements in [this proposal](https://github.com/godotengine/godot-proposals/issues/6681) (as "Add several tracks/keyframes when several nodes are selected").

This seemed fairly straight-forward to address, by simply allowing the keying to not only work on `Node` but also on `MultiNodeEdit`, which is exactly what this PR does.

One thing that's perhaps up for debate in terms of UX is what value should be put into the keys when you have multiple nodes selected, given that `MultiNodeEdit` will display the value from the first selected node, even if the values of the multiple nodes differ. It seemed to me that the most intuitive implementation was to have each node's respective value entered into their respective keys, rather than the value of the first selected node be entered into all nodes' keys. For this reason the `p_value` passed into `AnimationPlayerEditorPlugin::_property_keyed` is now ignored and fetched from each respective node instead.

(I would perhaps also argue that `MultiNodeEdit` shouldn't work this way to begin with, and should render the value/control in some other way when the values differ between the nodes, which I believe is how Unreal does it, and perhaps other editors as well, but that's a separate discussion.)

In the process of doing this I felt the need to create a new method in `AnimationTrackEditor` that would be similar to `insert_value_key` but which accepted a specific `Node` instead and found that there was already such a  method, called `insert_node_value_key`, so I made use of that instead (with some tweaks). The only major difference between these that I could spot was the inclusion of `make_insert_queue` and `commit_insert_queue` in the old `insert_value_key`, which I left in the now refactored `insert_value_key`. I welcome feedback as to whether this approach might have some unintended consequences.

### CODE REVIEW

**Comment by mihe** [comment]
I figured it made sense to have `insert_node_value_key` and the now refactored `insert_value_key` be consistent in terms of not taking a value but instead just fetching the value based on the property name passed in, so I did just that. It'll be a tad bit slower obviously, but it's not a hot path anyway, and this new interface should leave less room for shooting yourself in the foot by mismatching the property name and value.

File: editor/animation_track_editor.cpp [hunk 1]
```diff
@@ -4040,41 +4042,39 @@ void AnimationTrackEditor::insert_node_value_key(Node *p_node, const String &p_p
 
 	path += ":" + p_property;
 
-	NodePath np = path;
```
Comment (KoBeWi) [inline editor/animation_track_editor.cpp:4043]: Why did you remove it? Comparing NodePath with String is slower.

File: editor/animation_track_editor.cpp [hunk 2]
```diff
@@ -4100,107 +4100,43 @@ void AnimationTrackEditor::insert_node_value_key(Node *p_node, const String &p_p
...
-
-	path += ":" + p_property;
-
-	NodePath np = path;
-
-	// Locate track.
 
-	bool inserted = false;
+	MultiNodeEdit *multi_node_edit = Object::cast_to<MultiNodeEdit>(obj);
+	if (multi_node_edit != nullptr) {
```
Comment (KoBeWi) [inline editor/animation_track_editor.cpp:4121]: MultiNodeEdit is RefCounted, so this is a more standard way of handling it:
```suggestion
	Ref<MultiNodeEdit> multi_node_edit = obj;
	if (multi_node_edit.is_valid()) {
```

File: editor/animation_track_editor.cpp (hunk 1 above)
Comment (mihe) [reply editor/animation_track_editor.cpp]: My bad. It looked redundant at first glance. I've restored it now.

File: editor/animation_track_editor.cpp (hunk 2 above)
Comment (mihe) [reply editor/animation_track_editor.cpp]: That does seem to work, by virtue of `RefCounted::operator=(const Variant &)`, causing it to implicitly cast the pointer to a `Variant` first. I went ahead and made the suggested change.

Comment (KoBeWi) [reply editor/animation_track_editor.cpp]: Looks like it doesn't work. You can try `Ref<MultiNodeEdit> multi_node_edit(obj)` or `Ref<MultiNodeEdit> multi_node_edit = Ref<MultiNodeEdit>(obj);`.

Comment (mihe) [reply editor/animation_track_editor.cpp]: It's somewhat troubling that MSVC was happy to do the implicit `Variant` conversion but GCC was not. I wonder if it's because Godot isn't providing [`/permissive-`](https://learn.microsoft.com/en-us/cpp/build/reference/permissive-standards-conformance?view=msvc-170) when compiling with MSVC. Your latest suggestion seems to work in GCC as well though.

**Review by KoBeWi — APPROVED** [review]
Works correctly.

Could use a look from @TokageItLab

**Review by TokageItLab — APPROVED** [review]
Seems to be working well.

**Comment by akien-mga** [comment]
Thanks!
